compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -1119,7 +1119,13 @@ class Typer(@constructorOnly nestingLevel: Int = 0) extends Namer
           case Apply(fn, _) if fn.symbol.is(ExtensionMethod) =>
             def toSetter(fn: Tree): untpd.Tree = fn match
               case fn @ Ident(name: TermName) =>
-                untpd.cpy.Ident(fn)(name.setterName)
+                fn.tpe match
```

Comment:
It looks non-sensical that on the one hand we can refer to a lhs with a simple ident, but on the other hand the setter needs a full tree expansion. Can you add a comment why that is so?

Reply:
Added a comment